purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/PaywallData.kt

```diff
@@ -41,52 +43,60 @@ data class PaywallData(
     /**
      * Returns the [LocalizedConfiguration] to be used based on the current locale.
      */
-    val localizedConfiguration: LocalizedConfiguration
+    val localizedConfiguration: Pair<Locale, LocalizedConfiguration>
+        @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
         get() {
             val preferredLocales = LocaleListCompat.getDefault()
 
             for (i in 0 until preferredLocales.size()) {
                 preferredLocales.get(i)?.let { locale ->
-                    configForLocale(locale)?.let { localizedConfiguration ->
-                        return localizedConfiguration
+                    // This is a hack to make android studio previews work. In previews, the locale language method
+                    // gives a different result than expected (for example, "es-es" instead of "es").
+                    // In order to fix that, we convert the string to a locale and we parse that as a locale so the
+                    // language and country are correctly parsed.
+                    val localeToCheck = Locale.forLanguageTag(locale.toString())
+                    configForLocale(localeToCheck)?.let { localizedConfiguration ->
+                        return (localeToCheck to localizedConfiguration)
                     }
                 }
             }
 
             return fallbackLocalizedConfiguration
         }
 
-    private val fallbackLocalizedConfiguration: LocalizedConfiguration
+    private val fallbackLocalizedConfiguration: Pair<Locale, LocalizedConfiguration>
+        @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
         get() {
-            return localization.values.first()
+            localization.entries.first().let { localization ->
+                return (getLocaleForLabel(localization.key) to localization.value)
+            }
         }
 
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    private fun getLocaleForLabel(label: String): Locale {
+        // Parsing language requires it to be in the form "en-US", even though Locale.toString() returns "en_US"
+        return Locale.forLanguageTag(label.replace("_", "-"))
+    }
+
     /**
      * @note This allows searching by `Locale` with only language code and missing region (like `en`, `es`, etc).
      *
      * @return [LocalizedConfiguration] for the given [Locale], if found.
      */
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
     fun configForLocale(requiredLocale: Locale): LocalizedConfiguration? {
         return localization[requiredLocale.toString()]
             ?: localization.entries.firstOrNull { (localeKey, _) ->
-                @Suppress("UsePropertyAccessSyntax")
-                Locale(localeKey).languageWithFallback == requiredLocale.languageWithFallback
```

Comment:
This was wrong, since it could potentially end up comparing `isO3Language` with `language`

Reply:
right, good catch